ReactQt/runtime/src/reactuimanager.cpp

```diff
@@ -128,8 +128,8 @@ void ReactUIManager::setChildren
 {
   //TODO: This is a simple implementation which fixes a broken example. It's not properly tested and may need revisiting
   QList<int> indices;
-  foreach(int i, childrenTags) {
-    indices.append(0); //0 is a default z-order in qml
+  for(int i = 0; i < childrenTags.size(); ++i) {
+    indices.append(i); //Incrementing per each child. Similar to iOS. See the comment in manageChildren
```

Comment:
@narek-h , it seems we shouldn't set z-order on our side at all. (maybe for some not trivial hacks)

If JS code doesn't set some specific [zIndex](https://github.com/status-im/react-native-linux/blob/react-native-qt/Libraries/StyleSheet/LayoutPropTypes.js#L498) , we should use just 0 for zOrder, or to not set it at all, because 0 is default.

Objective-c implementation for iOS should give good example as it's done there.

Reply:
Hi @narek-h , I'm sorry, it seems I can't repro behavior that text is drawn at the top of cells with commented out setting of Z order and 0 indexes;
But if you take changes in your current PR ( indexes are incremented ) and just comment out setting of Z order, the text will drown at the top of cells.
So it seems that inserting into layout `rfl->insertChild(i, child);` in `ReactUIManager::manageChildren` reorders items from top to bottom. For me it's not clear why Z order is set here.

Reply:
@MaxRis , thanks for looking on that.  I agree it seems weird to use the index in layout to set Z orders.
Another funny thing is, if I use incremental indexes for layout insertion, and comment out setting z index in addChildItem method  ` vm->addChildItem(container, child, i);`  after pressing NewGame (second round)  the title gets lost/hidden  : ) 
I'll look on this and will try to find a solution where we don't set Z index at all, but everything is properly handled.  At this moment don't have such solution.